cores/esp8266/PolledTimeout.h

```diff
@@ -47,15 +47,40 @@ struct YieldOrSkip
 
 } //YieldPolicy
 
+namespace TimePolicy
+{
 
-template <bool PeriodicT, typename YieldPolicyT = YieldPolicy::DoNothing>  
+struct TimeMillis
+{
+  using timeType = decltype(millis());
+  static timeType time() {return millis();}
+  static constexpr timeType toMillis = 1;
+};
+
+#ifdef CORE_MOCK
+struct TimeCycle: public TimeMillis {};
+#else
+
+struct TimeCycle
+{
+  using timeType = decltype(ESP.getCycleCount());
+  static timeType time() {return ESP.getCycleCount();}
+  static constexpr timeType toMillis = F_CPU / 1000;
+};
+
+#endif // cpu cycles
+
+
+} //TimePolicy
+
+template <bool PeriodicT, typename YieldPolicyT = YieldPolicy::DoNothing, typename TimePolicyT = TimePolicy::TimeMillis>
 class timeoutTemplate
 {
 public:
-  using timeType = decltype(millis());
+  using timeType = typename TimePolicyT::timeType;
   
   timeoutTemplate(timeType timeout) 
```

Comment:
Given that you now implemented toMillis, the timeout constructor argument is now restricted to ms units. This merits a Big Phat Comment here.

Reply:
About specifying the timeout in ms:
My original intent was to not force a unit for the given timeout value, especially not ms.
Your intent seems to be higher precision, but the minimum argument value is 1ms, which equates to (I think) 160.000 cycles @160MHz. So you have high precision, but very low resolution.
Are you absolutely certain you want to force the constructor arg to be in ms?
Consider allowing the arg to be in number of cycles. You could then implement a static constexpr fromMillis(), which would then allow usage like this:
```
using oneShotCycles = esp8266::polledTimeout::oneShotCycles;
oneShotCycles timeout(oneShotCycles::fromMillis(500)); //500ms value gets converted to number of cycles and passed to the construction
```
With this, if a user wants the full per-cycle resolution, he can still provide a constructor arg in number of cycles.

Reply:
That wasn't the primary goal for this PR, I primarily needed `ms` resolution with a lighter test (in terms of cpu load).
But yes that's a fair notice, I'll add it as a third use case.